scripts/generators/es_template.py

```diff
@@ -20,29 +20,32 @@ def generate(ecs_flat, ecs_version):
 # Field mappings
 
 
-def dict_add_nested(dict, nestings, value):
+def dict_add_nested(dct, nestings, value):
     current_nesting = nestings[0]
     rest_nestings = nestings[1:]
     if len(rest_nestings) > 0:
-        if current_nesting not in dict:
-            dict[current_nesting] = {'properties': {}}
-        elif 'type' in dict[current_nesting] and 'object' == dict[current_nesting]['type']:
```

Comment:
We got lucky before, because with Python 2, the [ordering was deterministic](https://portingguide.readthedocs.io/en/latest/dicts.html#changed-key-order)

Reply:
The previous version was passing unit tests but generated code that was inconsistent, because `"type": "object"` was missing. I can make it so this is always present, but I don't exactly know the intendend behavior.
https://github.com/elastic/ecs/blob/94e7376ed1e618bec7128e4b14f579a5c0111343/generated/elasticsearch/7/template.json#L419-L496

Replacing these lines
https://github.com/elastic/ecs/blob/94e7376ed1e618bec7128e4b14f579a5c0111343/scripts/generators/es_template.py#L27-L30
with this should do what I _think_ is our intended behavior, but it's very clear to me what that is.
```python
        dct[current_nesting].setdefault("properties", {})
```

Reply:
Yeah iirc it's a nasty bit that I added when adding `dns.answers`. Adding `type: object` everywhere we have further nesting would be fine and a noop, as far as I understand.

I'd prefer to keep it as it was, and only handle the explicit `type: object` in place when it's already there, for now.